cvat-ui/src/components/annotation-page/canvas/grid-layout/canvas-layout.tsx

```diff
@@ -0,0 +1,357 @@
+        case ViewType.RELATED_IMAGE:
+            component = <ContextImage offset={offset} />;
+            break;
+        case ViewType.CANVAS_3D_FRONT:
+            component = <FrontViewComponent />;
+            break;
+        case ViewType.CANVAS_3D_SIDE:
+            component = <SideViewComponent />;
+            break;
+        case ViewType.CANVAS_3D_TOP:
+            component = <TopViewComponent />;
+            break;
+        default:
+            component = <div> Undefined view </div>;
+    }
+
+    return component;
+};
+
+const fitLayout = (type: DimensionType, layoutConfig: ItemLayout[], rows: number): ItemLayout[] => {
+    const updatedLayout: ItemLayout[] = [];
+
+    const relatedViews = layoutConfig
+        .filter((item: ItemLayout) => item.viewType === ViewType.RELATED_IMAGE);
+    const cols = relatedViews.length > 6 ? 2 : 1;
+    const height = Math.floor(rows / (relatedViews.length / cols));
+    relatedViews.forEach((view: ItemLayout, i: number) => {
+        updatedLayout.push({
+            ...view,
+            h: height,
+            w: relatedViews.length > 6 ? 2 : 3,
+            x: cols === 1 ? 9 : 8 + (i % 2) * 2,
+            y: height * i,
+        });
+    });
+
+    let widthAvail = 12;
+    if (relatedViews.length > 6) {
+        widthAvail = 8;
+    } else if (relatedViews.length > 0) {
+        widthAvail = 9;
+    }
+
+    if (type === DimensionType.DIM_2D) {
+        const canvas = layoutConfig
+            .find((item: ItemLayout) => item.viewType === ViewType.CANVAS) as ItemLayout;
+        updatedLayout.push({
+            ...canvas,
+            x: 0,
+            y: 0,
+            w: widthAvail,
+            h: 12,
+        });
+    } else {
+        const canvas = layoutConfig
+            .find((item: ItemLayout) => item.viewType === ViewType.CANVAS_3D) as ItemLayout;
+        const top = layoutConfig
+            .find((item: ItemLayout) => item.viewType === ViewType.CANVAS_3D_TOP) as ItemLayout;
+        const side = layoutConfig
+            .find((item: ItemLayout) => item.viewType === ViewType.CANVAS_3D_SIDE) as ItemLayout;
+        const front = layoutConfig
+            .find((item: ItemLayout) => item.viewType === ViewType.CANVAS_3D_FRONT) as ItemLayout;
+        updatedLayout.push({
+            ...canvas,
+            x: 0,
+            y: 0,
+            w: widthAvail,
+            h: 9,
+        }, {
+            ...top, x: 0, y: 9, w: Math.ceil(widthAvail / 3), h: 3,
+        },
+        {
+            ...side, x: Math.ceil(widthAvail / 3), y: 9, w: Math.ceil(widthAvail / 3), h: 3,
+        },
+        {
+            ...front,
+            x: Math.ceil(widthAvail / 3) * 2,
+            y: 9,
+            w: Math.floor(widthAvail / 3),
+            h: 3,
+        });
+    }
+
+    return updatedLayout;
+};
+
+function CanvasLayout({ type }: { type?: DimensionType }): JSX.Element {
+    const NUM_OF_ROWS = 12;
+    const MARGIN = 8;
+    const PADDING = MARGIN / 2;
```

Comment:
What do you think about moving these consts in config file? I see the same 12 in the function above